jwst/exp_to_source/exp_to_source.py

```diff
@@ -40,8 +40,26 @@ def exp_to_source(inputs):
             log.debug(f'Copying source {slit.source_id}')
             result_slit = result[str(slit.source_id)]
             result_slit.exposures.append(slit)
+            # store values for later use (after merge_tree)
+            # these values are incorrectly getting overwritten by
+            # the top model.
+            slit_bunit = slit.meta.bunit_data.copy()
```

Comment:
Some testing on a MultiSlitModel I have sitting around shows that @tapastro is right (are we surprised?). These first 3 objects (bunit_data, bunit_err, and model_type) are just strings and hence have no copy method (or instance for that matter). Plus, given the fact that they're simple strings, I doubt that these would be causing any kind of trouble when trying to insert their values later on into the new output model. So the remaining culprit might be `wcsinfo`? It too does NOT have a copy method, but you can effectively copy it using `wcsinfo.instance`. The "bare" `wcsinfo` is an object node in the input datamodel, but using `instance` turns it into a dict.